### DISCUSSION TITLE
Structuring Inner Blocks: Flat or Nested Approach?

### DISCUSSION THREAD

**Discussion by gigitux** [woocommerce/woocommerce#44964] [WooCommerce, Blocks, and Site Editing]
We are currently working on the Product Gallery block, which consists of a parent block with multiple children's blocks. To organize the code, we have created a parent folder named "product-gallery" and a subfolder within it called "inner-blocks" (refer to the provided image). All the inner blocks are located within this subfolder.

![image](https://github.com/woocommerce/woocommerce-blocks/assets/4463174/28377831-463c-4e73-b486-1d5c73f8942a)

However, we have encountered a question regarding the structure of the inner blocks. Since an inner block itself can be a parent of another block, is it necessary to create another "inner-blocks" folder within the folder of this latter block? Alternatively, we could consider a more flat structure with only a single "inner-blocks" folder.

The nested approach helps us to understand the relationship of inner blocks.
The single inner-blocks folder approach has a more clear hierarchy.

@tarhi-saad suggests this approach where the hierarchy is clear and the relationship is defined as well:
```
/inner-blocks/cart-order-summary
/inner-blocks/cart-order-summary-coupon
/inner-blocks/cart-order-summary-discount
```
The first one is an inner block and a parent block based on the naming convention! This approach is already used for the Cart & Checkout blocks.

What do you think?

**Comment by dinhtungdu** [comment]
```
/inner-blocks/cart-order-summary
/inner-blocks/cart-order-summary-coupon
/inner-blocks/cart-order-summary-discount
```

This looks optimal IMO. Having `inner-blocks` folder inside inner-block folders creates a maintainability issue and actually makes browsing source code harder for me.

**Comment by nielslange** [comment]
Personally, I also prefer a rather flat approach. @tarhi-saad's suggestion looks great to me.

**Comment by mikejolley** [comment]
I prefer nested so its clear which parent block owns the inner blocks. In cases where the inner blocks *must* have a certain parent block (think cart/empty cart/checkout) this makes that relationship clear.

Perhaps its actually structured based on that? Does an inner block have a specific parent, or can it have any parent (in which case its top level).

**Comment by tjcafferkey** [comment]
FWIW. I am of the same opinion as @mikejolley https://github.com/woocommerce/woocommerce-blocks/discussions/10188#discussioncomment-6437048
